### PR TITLE
Add New Package:PSCMC

### PR DESCRIPTION
**Pull request by Bitllion**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Did you intend to include the `abacus` package here?  These changes are already in #30933

**Comment by Bitllion** [comment]
> Did you intend to include the `abacus` package here? These changes are already in #30933

Sorry,This is a misunderstanding. 
I found a spack stytle errorr on abacus,
Reference to previous version：
`from spack import *`
I don't know when it was changed to this：
`from spack.package import *`
Now I have corrected the style error of the abacus branch, but the development branch is still wrong. Because the PSCM branch is just checked out from the development branch, it affects my pscmc branch

File: var/spack/repos/builtin/packages/pscmc/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/pscmc/package.py]:
```suggestion
# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please correct the ending copyright year.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**